Sources/AblyChat/Messages.swift

```diff
@@ -1,10 +1,35 @@
 import Ably
 
+public protocol MessageSubscriptionResponse: Sendable {
+    func unsubscribe() async -> Void
+    func getPreviousMessages() async throws -> any PaginatedResult<Message>
+}
+
+final public class MessageSubscriptionWrapper: Sendable, MessageSubscriptionResponse {
+    let unsubscribeCallback: @Sendable () async -> Void
+    let previousMessageCallback: @Sendable () async throws -> any PaginatedResult<Message>
+    
+    init(unsubscribe: @escaping @Sendable () async -> Void, getPreviousMessages: @escaping @Sendable () async throws -> any PaginatedResult<Message>) {
+        self.unsubscribeCallback = unsubscribe
+        self.previousMessageCallback = getPreviousMessages
+    }
+    
+    public func unsubscribe() async {
+        await unsubscribeCallback()
+    }
+    
+    public func getPreviousMessages() async throws -> any PaginatedResult<Message> {
+        try await previousMessageCallback()
+    }
+}
+
 public protocol Messages: AnyObject, Sendable, EmitsDiscontinuities {
-    func subscribe(bufferingPolicy: BufferingPolicy) -> MessageSubscription
+    func subscribe(bufferingPolicy: BufferingPolicy, listener: MessageEventPayloadWrapper) async throws -> MessageSubscriptionResponse
```

Comment:
What is this public API change — what is `listener` for? Have you decided to stop using `AsyncSequence` for messages?

Reply:
replaced with`MessageSubscription`